File: Easy/Python/349. Intersection of Two Arrays.py
'''
Given two integer arrays nums1 and nums2, return an array of their intersection. Each element in the result must be unique and you may return the result in any order.
'''

from typing import List

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        mapping = {}
        arr = []

        for num in nums1:
            if num not in mapping:
                mapping[num] = 1

        for num in nums2:
            if num in mapping:
                arr.append(num)
                del mapping[num]

        return arr

print(Solution().intersection([1,2,2,1],[2,2]))

File: Easy/Javascript/349. Intersection of Two Arrays.js
/**
 * Given two integer arrays nums1 and nums2, return an array of their intersection.
 * Each element in the result must be unique and you may return the result in any order.
 */


/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    let mapping = {};
    let arr = []

    for (let num of nums1){
        if (!(num in mapping)){
            mapping[num] = 1
        }
    }
    

    for (let num of nums2){
        if (num in mapping){
            arr.push(num);
            delete mapping[num]
        }
    }

    return arr
};

console.log(intersection([0],[0]))